src/Behat/Cores/Drupal7.php

```diff
@@ -177,21 +177,38 @@ public function getEditableConfig($name) {
    * {@inheritdoc}
    */
   public function getEntityByField($entity_type, $field_name, $value) {
-    throw new PendingException('Pending to implement method in Drupal 7');
+    $query = new \EntityFieldQuery();
+    $query->entityCondition('entity_type', $entity_type)
+      ->propertyCondition($field_name, $value)
+      ->entityOrderBy('entity_id', 'DESC')
+      ->range(0, 1);
+
+    $result = $query->execute();
+    $entity_wrapper = NULL;
+    if (!empty($result)) {
+      $user = reset($result);
+      $entity_id = array_keys($user)[0];
+      $entity_wrapper = entity_metadata_wrapper($entity_type, $entity_id);
+    }
+
+    return $entity_wrapper;
   }
 
   /**
    * {@inheritdoc}
    */
   public function getEntityFieldValue($field_name, $entity, $fallback = NULL) {
-    throw new PendingException('Pending to implement method in Drupal 7');
+    if (isset($entity->{$field_name})) {
```

Comment:
I'm not used to D7 anymore so I may ask something stupid, but are we sure $entity->{$field_name} will work for field names and also for properties like roles?

Reply:
Yes, it will work because we use an entity_metadata_wrapper in the method getEntityByField.
I tried it using the field roles and works.